.github/workflows/release-publish.yaml

```diff
@@ -36,25 +38,33 @@ jobs:
           path: dist/
   pypi:
     needs: ["source-wheel"]
-    runs-on: ubuntu-latest
+    runs-on: [self-hosted, jammy]
+    permissions:
+      # IMPORTANT: this permission is mandatory for trusted publishing
+      id-token: write
     steps:
       - name: Get packages
         uses: actions/download-artifact@v4
         with:
           name: pypi-packages
           path: dist/
       - name: Publish to pypi
+        # Note: this action uses PyPI's support for Trusted Publishers
+        # It needs a configuration on the PyPI project - see:
+        # https://docs.pypi.org/trusted-publishers/adding-a-publisher/#github-actions
```

Comment:
Basically:
- Nothing needs to be done on the Github repo side (that `PYPI_API_TOKEN` can be deleted);
- An admin on the PyPI project needs to follow the instructions on that link. It's simple:

![image](https://github.com/canonical/starbase/assets/1951299/29eb46c7-6c93-4434-b4dd-f864572c72b9)